lib/migration.pm

```diff
@@ -143,6 +144,13 @@ sub disable_installation_repos {
     }
 }
 
+# Based on bsc#1097111, need to disable kernel multiversion before migration, and enable it after migration
+# https://documentation.suse.com/sles/15-SP3/html/SLES-all/cha-update-preparation.html#sec-update-preparation-multiversion 
+sub disable_kernel_multiversion_temporarily {
+    my $sed_para = check_var('VERSION', get_var('ORIGIN_SYSTEM_VERSION')) && check_var('VERSION', get_var('UPGRADE_TARGET_VERSION')) ? 's/^multiversion/#multiversion/g' : 's/^#multiversion/multiversion/g';
+    script_run('sed -i $sed_para /etc/zypp/zypp.conf');
```

Comment:
I think you can move the record_soft_failure into the function.
And normally the record_soft_failure should be 'bug ID' , 'Bug title'.